### PR TITLE
feat: Mailchimp

### LINKED ISSUE #1130: [DF-500] Implement a mailchimp provider
Implement support for [Mailchimp](https://mailchimp.com/) and mandrill as an email service provider.

Add a new `PersistedEmailProvider` type:

[https://github.com/dittofeed/dittofeed/blob/098d153f764da44f4df48b8930f97a431f240883/packages/isomorphic-lib/src/types.ts#L1465-L1472](https://github.com/dittofeed/dittofeed/blob/098d153f764da44f4df48b8930f97a431f240883/packages/isomorphic-lib/src/types.ts#L1465-L1472)

Add new `EmailServiceProviderSuccess` type:

[https://github.com/dittofeed/dittofeed/blob/098d153f764da44f4df48b8930f97a431f240883/packages/isomorphic-lib/src/types.ts#L3068-L3075](https://github.com/dittofeed/dittofeed/blob/098d153f764da44f4df48b8930f97a431f240883/packages/isomorphic-lib/src/types.ts#L3068-L3075)

Add new `EmailServiceProviderFailure` type:

[https://github.com/dittofeed/dittofeed/blob/098d153f764da44f4df48b8930f97a431f240883/packages/isomorphic-lib/src/types.ts#L3275-L3281](https://github.com/dittofeed/dittofeed/blob/098d153f764da44f4df48b8930f97a431f240883/packages/isomorphic-lib/src/types.ts#L3275-L3281)

Add new `EmailProviderSecret` type:

[https://github.com/dittofeed/dittofeed/blob/098d153f764da44f4df48b8930f97a431f240883/packages/isomorphic-lib/src/types.ts#L3610-L3617](https://github.com/dittofeed/dittofeed/blob/098d153f764da44f4df48b8930f97a431f240883/packages/isomorphic-lib/src/types.ts#L3610-L3617)

Error handling for the `/templates/test` api endpoint:

[https://github.com/dittofeed/dittofeed/blob/098d153f764da44f4df48b8930f97a431f240883/packages/api/src/controllers/contentController.ts#L418-L437](https://github.com/dittofeed/dittofeed/blob/098d153f764da44f4df48b8930f97a431f240883/packages/api/src/controllers/contentController.ts#L418-L437)

A webhook for updating the status of sent emails:

[https://github.com/dittofeed/dittofeed/blob/098d153f764da44f4df48b8930f97a431f240883/packages/api/src/controllers/webhooksController.ts#L68-L80](https://github.com/dittofeed/dittofeed/blob/098d153f764da44f4df48b8930f97a431f240883/packages/api/src/controllers/webhooksController.ts#L68-L80)

Handle the email provider type when sending messages:

[https://github.com/dittofeed/dittofeed/blob/098d153f764da44f4df48b8930f97a431f240883/packages/backend-lib/src/messaging.ts#L899-L909](https://github.com/dittofeed/dittofeed/blob/098d153f764da44f4df48b8930f97a431f240883/packages/backend-lib/src/messaging.ts#L899-L909)

Create the email provider:

[https://github.com/dittofeed/dittofeed/blob/098d153f764da44f4df48b8930f97a431f240883/packages/backend-lib/src/messaging/email.ts#L143-L152](https://github.com/dittofeed/dittofeed/blob/098d153f764da44f4df48b8930f97a431f240883/packages/backend-lib/src/messaging/email.ts#L143-L152)

Add Place in Settings to configure email provider:

[https://github.com/dittofeed/dittofeed/blob/098d153f764da44f4df48b8930f97a431f240883/packages/dashboard/src/pages/settings.page.tsx#L1060-L1072](https://github.com/dittofeed/dittofeed/blob/098d153f764da44f4df48b8930f97a431f240883/packages/dashboard/src/pages/settings.page.tsx#L1060-L1072)

See this previous issue as a reference: [https://github.com/dittofeed/dittofeed/issues/641](https://github.com/dittofeed/dittofeed/issues/641)

<sub>From [SyncLinear.com](https://synclinear.com) | [DF-500](https://linear.app/dittofeed/issue/DF-500/implement-a-mailchimp-provider)</sub>

### PR DESCRIPTION
**Pull request by neo773**
/closes #1130 
/claim #1130

### CODE REVIEW

**Review by maxgurewitz — CHANGES_REQUESTED** [review]
Some requested changes, also there are some linter errors!

File: packages/backend-lib/src/messaging.ts [hunk 1]
```diff
@@ -1224,6 +1226,77 @@ export async function sendEmail({
...
+    case EmailProviderType.MailChimp: {
+      const mailData: MessagesMessage = {
+        html: body,
+        text: body,
+        subject,
+        from_email: from,
+        to: [{ email: to }],
+        metadata: {
+          user_id: userId,
+          website: "https://courier.com", // TODO: Add a default website or pass it from configuration
```
[the remark this answers is not in the document]
Comment (neo773) [reply packages/backend-lib/src/messaging.ts]: Yes, it's required by the library

[the remark this answers is not in the document]
Comment (maxgurewitz) [reply packages/backend-lib/src/messaging.ts]: Can you change it to dittofeed.com then?

[the remark this answers is not in the document]
Comment (neo773) [reply packages/backend-lib/src/messaging.ts]: diff is outdated, we're deriving the website from sender email

**Review by maxgurewitz — APPROVED** [review]
Tested it out, webhooks working!

**Pull request merged**